metacat-main/src/main/java/com/netflix/metacat/main/services/search/ElasticSearchMigrationUtil.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.netflix.metacat.main.services.search;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.Lists;
+import com.netflix.metacat.common.MetacatRequestContext;
+import com.netflix.metacat.common.json.MetacatJson;
+import com.netflix.metacat.common.server.Config;
+import org.elasticsearch.client.Client;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * Class for migrating the elastic search metacat index to a new one. During the migration period, metacat
+ * writes and updates the data to both the existing and the new merge index. This class overrides the
+ * save, update, and softdelete methods. For the save op, it saves to bothe indices. for the update and softdelete,
+ * it first executes the normal index ops, then copies to the merge index
+ */
+public class ElasticSearchMigrationUtil extends ElasticSearchUtil {
```

Comment:
This may be a stupid statement but it seems to me this class really acts more like an aspect where when enabled it needs to perform some added action above and beyond the original action in the base class. That said why don't we just make an aspect where after every call on the core ElasticSearchUtil an aspect is enabled (when the migration is enabled) to invoke the copy from the base index to the new index. Basically almost a retry that does the same actions on the core index with a secondary index. Or just re-write the core elasticsearch util to allow you to send in the write indices and then iterate through them for all the methods in there at runtime. Then you just change the write-indices using a setter when the migration is active and remove the old index from the list once the migration is done. This would save from maintaining two classes and overriding methods, etc.